CMakeLists.txt

```diff
@@ -63,75 +111,51 @@ add_library(fossilize STATIC
-    endif()
-endif()
-
-target_include_directories(fossilize PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_compile_definitions(fossilize
+  PUBLIC
+    $<$<CXX_COMPILER_ID:MSVC>:_CRT_SECURE_NO_WARNINGS>
+    $<$<CXX_COMPILER_ID:MSVC>:_SCL_SECURE_NO_WARNINGS>
+    $<$<CXX_COMPILER_ID:MSVC>:NOMINMAX>
+    $<$<AND:$<PLATFORM_ID:Windows>,$<CXX_COMPILER_ID:GNU>>:__USE_MINGW_ANSI_STDIO=1>)
+target_sources(fossilize
+  PRIVATE
+    $<$<NOT:$<PLATFORM_ID:Android>>:fossilize_external_replayer.cpp>
+    $<$<NOT:$<PLATFORM_ID:Android>>:fossilize_external_replayer.hpp>
+    $<$<NOT:$<PLATFORM_ID:Windows>>:fossilize_external_replayer_linux.hpp>
+    $<$<PLATFORM_ID:Windows>:fossilize_external_replayer_windows.hpp>
+    $<$<PLATFORM_ID:Darwin>:platform/gcc_clang_spinlock.hpp>
+    $<$<PLATFORM_ID:Linux>:platform/futex_wrapper_linux.hpp>)
+target_include_directories(fossilize
+  PUBLIC
+    $<BUILD_INTERFACE:$<$<PLATFORM_ID:Windows>:${PROJECT_SOURCE_DIR}/cli/dirent/include>>
+    $<BUILD_INTERFACE:${FOSSILIZE_RAPIDJSON_INCLUDE_PATH}>
+    $<BUILD_INTERFACE:${FOSSILIZE_VULKAN_INCLUDE_PATH}>
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 target_compile_options(fossilize PRIVATE ${FOSSILIZE_CXX_FLAGS})
-if (NOT WIN32)
-    target_link_libraries(fossilize -pthread)
-    target_compile_options(fossilize PUBLIC -pthread)
-    if (NOT APPLE)
-        if (ANDROID)
-            target_link_libraries(fossilize android log)
-        else()
-            target_link_libraries(fossilize rt)
-        endif()
-    endif()
-endif()
-target_link_libraries(fossilize miniz)
-
-if (WIN32)
-    target_include_directories(fossilize PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/cli/dirent/include)
-endif()
-
-option(FOSSILIZE_VULKAN_LAYER "Build Vulkan layer." ON)
-if (FOSSILIZE_VULKAN_LAYER)
-    set_target_properties(fossilize PROPERTIES POSITION_INDEPENDENT_CODE ON)
-endif()
-
-option(FOSSILIZE_RAPIDJSON_INCLUDE_PATH "Optional custom include path for rapidjson.")
-if (FOSSILIZE_RAPIDJSON_INCLUDE_PATH)
-    target_include_directories(fossilize PUBLIC ${FOSSILIZE_RAPIDJSON_INCLUDE_PATH})
+target_link_libraries(fossilize
+  PRIVATE
+    $<TARGET_NAME_IF_EXISTS:Vulkan::Vulkan>
+    Threads::Threads
+    $<$<PLATFORM_ID:Android>:android>
+    $<$<PLATFORM_ID:Android>:log>
+    $<$<PLATFORM_ID:Linux>:rt>
+    miniz)
+
+if (CMAKE_VERSION LESS 3.13)
+  set_property(TARGET fossilize APPEND_STRING PROPERTY LINK_FLAGS ${FOSSILIZE_LINK_FLAGS})
 else()
-    target_include_directories(fossilize PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/rapidjson/include)
-endif()
-
-option(FOSSILIZE_VULKAN_INCLUDE_PATH "Optional custom include path for Vulkan headers.")
-if (FOSSILIZE_VULKAN_INCLUDE_PATH)
-    target_include_directories(fossilize PUBLIC ${FOSSILIZE_VULKAN_INCLUDE_PATH})
-else()
-    target_include_directories(fossilize PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/khronos/vulkan)
+  target_link_options(fossilize PRIVATE ${FOSSILIZE_LINK_FLAGS})
```

Comment:
This fails Travis.

Reply:
Is it this line in particular, or some other issue? I think I might have messed up the stringification here, which I can put into another commit to fix the above changes.

Reply:
```
CMake Error at CMakeLists.txt:148 (target_link_options):

  Unknown CMake command "target_link_options".
```

Reply:
I'll have to do some work to get this to work correctly. It shouldn't have entered the `else()` block and parsed this. Regardless, I needed to make a proper shim anyhow, and this was just a band-aid.

Reply:
This turned out to be a typo on my part. The build succeeds in configuring (but fails tests otherwise it would seem 😅)